pkg/api/validate.go

```diff
@@ -501,3 +518,10 @@ func (v *Validator) validateProvisioningState(ps ProvisioningState) (errs []erro
 	}
 	return
 }
+
+func (v *Validator) validateUpdateConfig(internalConfig, adminConfig Config) (errs []error) {
+	if !reflect.DeepEqual(internalConfig, adminConfig) {
```

Comment:
wouldn't this be a problem for key rotation which needs to have certain fields in the Config invalidated and regenerated before using that in the update?

Reply:
Yes, we agreed that in the first cut of this, we should invalidate every change. Your changes should enable validating certificate updates.